itou/www/apply/views/submit_views.py

```diff
@@ -546,32 +546,22 @@ def setup(self, request, *args, **kwargs):
 
         self.form = CreateOrUpdateJobSeekerStep3Form(
             data=request.POST or None,
-            initial=self.job_seeker_session.get("profile", {})
-            | {
+            # TODO(xfernandez): drop fallback to user dict from job_seeker_session
+            # in a few weeks
+            initial={
                 "pole_emploi_id": self.job_seeker_session.get("user").get("pole_emploi_id"),
                 "lack_of_pole_emploi_id_reason": self.job_seeker_session.get("user").get(
                     "lack_of_pole_emploi_id_reason"
                 ),
-            },
+            }
+            | self.job_seeker_session.get("profile", {}),
         )
 
     def post(self, request, *args, **kwargs):
         if self.form.is_valid():
             self.job_seeker_session.set(
                 "profile",
-                {
-                    k: v
-                    for k, v in self.form.cleaned_data.items()
-                    if k not in ["pole_emploi_id", "lack_of_pole_emploi_id_reason"]
-                },
-            )
-            self.job_seeker_session.set(
-                "user",
-                self.job_seeker_session.get("user")
-                | {
-                    "pole_emploi_id": self.form.cleaned_data.get("pole_emploi_id"),
-                    "lack_of_pole_emploi_id_reason": self.form.cleaned_data.get("lack_of_pole_emploi_id_reason"),
-                },
+                {k: v for k, v in self.form.cleaned_data.items()},
```

Comment:
Ça peux pas se simplifier en ça ?
```suggestion
                self.form.cleaned_data,
```